visualization/dashboard/src/MLIDashboard/Controls/FilterEditor.tsx

```diff
@@ -217,14 +344,29 @@ export class FilterEditor extends React.PureComponent<IFilterEditorProps, IFilte
     }
 
     private readonly setAsCategorical = (ev: React.FormEvent<HTMLElement>, checked: boolean): void => {
-        this.props.jointDataset.setTreatAsCategorical(this.state.column, checked);
+        const openedFilter = this.state.openedFilter;
+        this.props.jointDataset.setTreatAsCategorical(openedFilter.column, checked);
         if (checked) {
-            this.setState({arg:[], method: FilterMethods.includes});
+          this.setState(
+              {
+                  openedFilter:{
+                      arg:[],
+                      method:FilterMethods.includes,
+                      column:openedFilter.column
+                  }
+              }
+          )
         } else {
-            this.setState({
-                arg:this.props.jointDataset.metaDict[this.state.column].featureRange.max,
-                method: FilterMethods.lessThan
-            });
+
+            this.setState(
+                {
+                    openedFilter:{
+                        arg:this.props.jointDataset.metaDict[openedFilter.column].featureRange.max,
+                        method:FilterMethods.includes,
```

Comment:
this is why its breaking on switching back. Set back to numeric filter properties, with method is lessThan, and keeping column: openFilter.column